seqr/views/apis/saved_variant_api.py

```diff
@@ -72,110 +61,180 @@ def create_saved_variant_handler(request):
-        )
+    for variant_guid in variant_guids:
+        saved_variant = SavedVariant.objects.get(guid=variant_guid)
+        check_permissions(saved_variant.family.project, request.user, CAN_VIEW)
 
-    gene_note = {gene_id: {
-        'notes': get_json_for_gene_notes_by_gene_id([gene_id], request.user).get(gene_id, [])}} if save_as_gene_note else {}
+        if save_as_gene_note:
+            main_transcript_id = saved_variant.selected_main_transcript_id or saved_variant.saved_variant_json['mainTranscriptId']
+            gene_id = next(
+                (gene_id for gene_id, transcripts in saved_variant.saved_variant_json['transcripts'].items()
+                 if any(t['transcriptId'] == main_transcript_id for t in transcripts)), None) if main_transcript_id else None
+            create_seqr_model(
+                GeneNote,
+                note=request_json.get('note'),
+                gene_id=gene_id,
+                created_by=request.user,
+            )
 
-    _create_variant_note(saved_variant, request_json, request.user)
+        gene_note = {gene_id: {
+            'notes': get_json_for_gene_notes_by_gene_id([gene_id], request.user).get(gene_id, [])}} if save_as_gene_note else {}
 
-    variant_note = {variant_guid: {
-        'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]},
-    }
+        saved_variants.append(saved_variant)
+
+    _create_variant_note(saved_variants, request_json, request.user)
+
+    variant_note = {}
+    for variant_guid in variant_guids:
+        variant_note[variant_guid] = {'notes': [get_json_for_variant_note(tag) for tag in saved_variant.variantnote_set.all()]}
 
     return create_json_response({
         'savedVariantsByGuid': variant_note,
         'genesById': gene_note
     })
 
 
-def _create_variant_note(saved_variant, note_json, user):
+def _create_variant_note(saved_variants, note_json, user):
     create_seqr_model(
         VariantNote,
-        saved_variant=saved_variant,
         note=note_json.get('note'),
         submit_to_clinvar=note_json.get('submitToClinvar') or False,
         search_hash=note_json.get('searchHash'),
         created_by=user,
-    )
+    ).saved_variants.add(*saved_variants)
 
 
 @login_required(login_url=API_LOGIN_REQUIRED_URL)
 @csrf_exempt
-def update_variant_note_handler(request, variant_guid, note_guid):
-    saved_variant = SavedVariant.objects.get(guid=variant_guid)
-    check_permissions(saved_variant.family.project, request.user, CAN_VIEW)
-    note = VariantNote.objects.get(guid=note_guid, saved_variant=saved_variant)
+def update_variant_note_handler(request, variant_guids, note_guid):
+    variant_guids = variant_guids.split(',')
+    saved_variants = []
+    for variant_guid in variant_guids:
+        saved_variant = SavedVariant.objects.get(guid=variant_guid)
+        check_permissions(saved_variant.family.project, request.user, CAN_VIEW)
+        saved_variants.append(saved_variant)
```

Comment:
You can do this as one query with saved_variants = SavedVariant.objects.get(guid__in= variant_guids)